### ISSUE TITLE
`Axes3DSubplot.plot` not setting data limits correctly when `zdir='x'` or `zdir='y'`

### ISSUE THREAD

**Issue by goretkin** [matplotlib/matplotlib#5742]
On 1.5.0

``` python
import matplotlib.pyplot as plt
from mpl_toolkits.mplot3d import Axes3D
import numpy as np
ax = Axes3D(plt.gcf())

xs = np.r_[np.linspace(-1,1,100), -1]
ys1 = np.cos(6*xs)
ys2 = np.cos(6*xs+np.pi/3)
ys3 = np.cos(6*xs+2*np.pi/3)

ax.plot(xs, ys1, zs=0, zdir='x')
ax.plot(xs, ys2, zs=0, zdir='y')
ax.plot(xs, ys3, zs=0, zdir='z')
```

produces
![mpl_issue_bad](https://cloud.githubusercontent.com/assets/581043/11997831/6c6b329a-aa56-11e5-95fb-19f098fa9b4c.png)

but I expect it to produce this:
![mpl_issue_good](https://cloud.githubusercontent.com/assets/581043/11997834/8bcbe9ae-aa56-11e5-9107-386300bcef11.png)
(which I achieved by manually setting the limits)

**Comment by WeatherGod** [comment]
Hmmm, i wonder if the auto scaling is getting confused by the use of zdir?
I'll have to take a look into this later this week.

**Comment by tacaswell** [comment]
@goretkin Did this ever work?

**Comment by goretkin** [comment]
I don't know. I am pretty sure I have never tried it before. Do you suspect
it's a regression?

**Comment by mdboom** [comment]
FWIW: It seems to be broken at least as far back as 1.3.0

**Comment by WeatherGod** [comment]
Closed by #10728
